### DISCUSSION TITLE
[Notify] Expose a QNotifyOptions interface for the Notify plugin similar to QDialogOptions

### DISCUSSION THREAD

**Discussion by Basaingeal** [quasarframework/quasar#12877] [Ideas / Proposals] [answered]
The Dialog plugin has a nice `QDialogOptions` interface that I can use if I want to abstract dialog creation into different Vue hooks. 

The options for `notify(opts)` are just defined as part of the notification function type, which makes it difficult to type any kind of options object that will eventually be passed into `notify` but not immediately. 

It would be nice to have a `QNotifyOptions` interface to use, just how there is already one for the Dialog plugin.

**Comment by Basaingeal** [comment]
This is what I'm doing currently to export the type.

```typescript
export type QNotifyOptions = Extract<
  Parameters<typeof Notify.create>[0],
  { message?: string | undefined }
>;
```

**Comment by rstoenescu** [comment]
@yusufkandemir care to implement this?

**Reply by yusufkandemir** [reply]
Done #12962

**Comment by yusufkandemir** [comment, marked as answer]
Improvement will be available on the next Quasar release, once #12962 gets merged.

**Reply by yusufkandemir** [reply]
The improvement is available on https://github.com/quasarframework/quasar/releases/tag/quasar-v2.6.3
